properties/base/base.py

```diff
@@ -436,12 +436,51 @@ def equal(self, other):
         Equivalence is determined by checking if all Property values on
         two instances are equal, using :code:`Property.equal`.
         """
-        if self is other:
-            return True
-        if not isinstance(other, self.__class__):
-            return False
-        for prop in itervalues(self._props):
-            if prop.equal(getattr(self, prop.name), getattr(other, prop.name)):
-                continue
-            return False
+        warn('HasProperties.equal has been depricated in favor of '
+             'properties.equal and will be removed in the next release',
+             FutureWarning)
+        return equal(self, other)
+
+
+@utils.stop_recursion_with(False)
+def equal(value_a, value_b):
+    """Determine if two **HasProperties** instances are equivalent
+
+    Equivalence is determined by checking if (1) the two instances are
+    the same class and (2) all Property values on two instances are
+    equal, using :code:`Property.equal`. If the two values are the same
+    HasProperties instance (eg. :code:`value_a is value_b`) this method
+    returns True. Finally, if either value is not a HasProperties
+    instance, equality is simply checked with ==.
+    """
+    if (
+            not isinstance(value_a, HasProperties) or
+            not isinstance(value_b, HasProperties)
+    ):
+        return value_a == value_b
+    if value_a is value_b:
         return True
+    if value_a.__class__ is not value_b.__class__:
+        return False
+    for prop in itervalues(value_a._props):
```

Comment:
@bsmithyman - You are absolutely right about this precedent. However, I think of `HasProperties` instances more as objects, and the precedent there is that different objects are not the same even if they look the same (ie `(object() in [object()]) is False`). This philosophy is not necessarily correct, however, or at least, the philosophy could change depending on circumstance.

I guess what I mean is: Are HasProperties instances representations of actual objects or just representations of the _mutable state_ of objects? I think probably the best option is to allow for both interpretations, possibly even including a `HasState` class (or something) with `properties`. This would mean refactoring the code a bit to allow for both philosophies. I made an issue on this here: https://github.com/3ptscience/properties/issues/150

For now, I will keep this as-is, with the object-like interpretation.